src/main/java/net/datafaker/providers/base/TimeAndDate.java

```diff
@@ -0,0 +1,334 @@
+     * @param referenceDate the past date relative to this date.
+     * @param pattern       date time pattern to convert to string.
+     * @return a string representation of a past date relative to {@code referenceDate}.
+     */
+    public String past(int atMost, TimeUnit unit, Instant referenceDate, String pattern) {
+        return formatInstant(past(atMost, unit, referenceDate), pattern);
+    }
+
+    /**
+     * Generates a random date between two dates.
+     *
+     * @param from the lower bound inclusive
+     * @param to   the upper bound exclusive
+     * @return a random date between {@code from} and {@code to}.
+     * @throws IllegalArgumentException if the {@code to} date represents an earlier date than {@code from} date.
+     */
+    public Instant between(Instant from, Instant to) throws IllegalArgumentException {
+        if (to.isBefore(from)) {
+            throw new IllegalArgumentException("Invalid date range, the upper bound date is before the lower bound.");
+        }
+
+        if (from.equals(to)) {
+            return from;
+        }
+
+        long offsetMillis = faker.random().nextLong(to.toEpochMilli() - from.toEpochMilli());
+        return from.plusMillis(offsetMillis);
+    }
+
+    /**
+     * Generates a string representation of a random date between two dates.
+     *
+     * @param from    the lower bound inclusive
+     * @param to      the upper bound exclusive
+     * @param pattern date time pattern to convert to string.
+     * @return a string representation of a random date between {@code from} and {@code to}.
+     * @throws IllegalArgumentException if the {@code to} date represents an earlier date than {@code from} date.
+     */
+    public String between(Instant from, Instant to, String pattern) throws IllegalArgumentException {
+        return formatInstant(between(from, to), pattern);
+    }
+
+    /**
+     * Generates a random birthday between 65 and 18 years ago from now.
+     *
+     * @return a random birthday between 65 and 18 years ago from now.
+     */
+    public LocalDate birthday() {
+        return birthday(DEFAULT_MIN_AGE, DEFAULT_MAX_AGE);
+    }
+
+    /**
+     * Generates a string representation of a random birthday between 65 and 18 years ago from now.
+     *
+     * @param pattern date time pattern to convert to string.
+     * @return a string representation of a random birthday between 65 and 18 years ago from now.
+     */
+    public String birthday(String pattern) {
+        return formatInstant(birthday(DEFAULT_MIN_AGE, DEFAULT_MAX_AGE), pattern);
+    }
+
+    /**
+     * Generates a random birthday between two ages from now.
+     *
+     * @param minAge the minimal age
+     * @param maxAge the maximal age
+     * @return a random birthday between {@code minAge} and {@code maxAge} years ago from now.
+     * Negative {@code minAge} and {@code maxAge} are supported.
+     */
+    public LocalDate birthday(int minAge, int maxAge) {
```

Comment:
Make sure they're not negative or zero?

Reply:
> Make sure they're not negative or zero?

Not sure, since "Negative {@code minAge} and {@code maxAge} are supported.". ?

Reply:
Okay, wasn't sure that was intentional.